src/utils/record-completeness/ComputeRecordCompleteness.ts

```diff
@@ -1,92 +1,113 @@
-  if (vehicleType !== VEHICLE_TYPE.TRL) {
-    return validateRecordCompleteness(coreMandatoryValidation.psvHgvCarLgvMotoCoreMandatoryVehicleAttributes, vehicleAttributes);
+  const generalErrors = validateVehicleAttributes(vehicle)?.error;
+
+  if (generalErrors) {
+    console.log("general errors: ",generalErrors);
+    return RECORD_COMPLETENESS_ENUM.SKELETON;
+  }
+
+  const mandatoryAttributesValidationResult = validateMandatoryTechRecordAttributes(vehicle);
+
+  const isCoreMandatoryValid = !mandatoryAttributesValidationResult.coreValidationResult?.error;
+
+  const isNonCoreMandatoryValid = !mandatoryAttributesValidationResult.nonCoreValidationResult?.error;
+
+  if (isCoreMandatoryValid && isNonCoreMandatoryValid) {
+    return RECORD_COMPLETENESS_ENUM.COMPLETE;
+  } else if (isCoreMandatoryValid) {
+    return RECORD_COMPLETENESS_ENUM.TESTABLE;
   } else {
-    return validateRecordCompleteness(coreMandatoryValidation.trlCoreMandatoryVehicleAttributes, vehicleAttributes);
+    return RECORD_COMPLETENESS_ENUM.SKELETON;
   }
 };
 
-const validateCoreAndNonCoreMandatoryTechRecordAttributes = (vehicleType: string, techRecordWrapper: Vehicle) => {
-  let coreMandatoryValidationResult;
-  let nonCoreMandatoryValidationResult;
-  let coreMandatorySchema: ObjectSchema;
-  let nonCoreMandatorySchema: ObjectSchema | undefined;
-  let techRecord = techRecordWrapper.techRecord[0];
-  if (vehicleType === VEHICLE_TYPE.HGV) {
-    coreMandatorySchema = coreMandatoryValidation.hgvCoreMandatorySchema;
-    nonCoreMandatorySchema = nonCoreMandatoryValidation.hgvNonCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.PSV) {
-    coreMandatorySchema = coreMandatoryValidation.psvCoreMandatorySchema;
-    nonCoreMandatorySchema = nonCoreMandatoryValidation.psvNonCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.TRL) {
-    techRecord = {...techRecord, primaryVrm: techRecordWrapper.primaryVrm} as any;
-    coreMandatorySchema = coreMandatoryValidation.trlCoreMandatorySchema;
-    nonCoreMandatorySchema = nonCoreMandatoryValidation.trlNonCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.LGV) {
-    coreMandatorySchema = coreMandatoryValidation.lgvCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.CAR) {
-    coreMandatorySchema = coreMandatoryValidation.carCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.MOTORCYCLE) {
-    coreMandatorySchema = coreMandatoryValidation.motorcycleCoreMandatorySchema;
+function validateVehicleAttributes(vehicle: Vehicle): ValidationResult | undefined {
+  const vehicleType = vehicle.techRecord[0].vehicleType;
+
+  const validationSchema = vehicleType === VEHICLE_TYPE.TRL
+    ? trlCoreMandatoryVehicleAttributes
+    : psvHgvCarLgvMotoCoreMandatoryVehicleAttributes
+
+  if (!vehicleType) {
+    return missingVehicleTypeValidationResult;
+  } else if (!validationSchema) {
+    return undefined;
```

Comment:
Is this code reachable? `validationSchema` will always be defined if `vehicleType` is defined

Reply:
+1 agree not reachable code

Reply:
The code we inherited was doing the same check in case `validationSchema` would be `undefined`. So, I followed their pattern. If it is not a big problem, I would like to keep this just to be on the safe side.